src/architecture/_GeneralModuleFiles/swig_conly_data.i

```diff
@@ -205,11 +215,82 @@ ARRAYINTASLIST(unsigned int)
 }
 
 %enddef
-
 ARRAY2ASLIST(double)
-ARRAY2ASLIST(int)
 ARRAY2ASLIST(float)
-ARRAY2ASLIST(unsigned int)
+
+%define ARRAYINT2ASLIST(type, fromfunc, asfunc)
+
+%typemap(in) type [ANY][ANY] (type temp[$1_dim0][$1_dim1]) {
+    Py_ssize_t i;
+    if(!PySequence_Check($input)) {
+        PyErr_SetString(PyExc_ValueError,"Expected a list of lists!  Does not appear to be that.");
+        return NULL;
+    }
+    Py_ssize_t rowLength = 0;
+    for(i=0; i<PySequence_Length($input); i++){
+        PyObject *obj = PySequence_GetItem($input, i);
+        if(!PySequence_Check($input)) {
+            printf("Row bad in matrix: %zd\n", i);
+            PyErr_SetString(PyExc_ValueError,"Need a list for each row");
+        }
+        if(!rowLength)
+        {
+            rowLength = PySequence_Length(obj);
+        }
+        if(!rowLength || rowLength != PySequence_Length(obj))
+        {
+            printf("Row bad in matrix: %zd\n", i);
+            PyErr_SetString(PyExc_ValueError,"All rows must be the same length!");
+        }
+        int j;
+        for(j=0; j<rowLength; j++)
+        {
+            temp[i][j] = (type)asfunc(PySequence_GetItem(obj, j));
```

Comment:
I think the object you get when calling `PySequence_GetItem(obj, j)` is leaking.